source/sites/publicodes/Simulateur.tsx

```diff
@@ -0,0 +1,339 @@
+import { goToQuestion, setSimulationConfig } from '@/actions/actions'
+import { getRelatedMosaicInfosIfExists } from '@/components/conversation/RuleInput'
+import {
+	Category,
+	DottedName,
+	extractCategories,
+	FullName,
+	isMosaicChild,
+	isRootRule,
+	MODEL_ROOT_RULE_NAME,
+	NGCRulesNodes,
+} from '@/components/publicodesUtils'
+import { buildEndURL } from '@/components/SessionBar'
+import Simulation from '@/components/Simulation'
+import Title from '@/components/Title'
+import { useEngine } from '@/components/utils/EngineContext'
+import { Markdown } from '@/components/utils/markdown'
+import Meta from '@/components/utils/Meta'
+import { AppState } from '@/reducers/rootReducer'
+import { motion } from 'framer-motion'
+import { utils } from 'publicodes'
+import { useEffect } from 'react'
+import { Trans } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { Navigate, useNavigate } from 'react-router'
+import { Link, useParams } from 'react-router-dom'
+import BandeauContribuer from './BandeauContribuer'
+import InlineCategoryChart from './chart/InlineCategoryChart'
+import { enquêteSelector } from './enquête/enquêteSelector'
+import { questionConfig } from './questionConfig'
+import ScoreBar from './ScoreBar'
+
+const equivalentTargetArrays = (array1, array2) =>
+	array1.length === array2.length &&
+	array1.every((value, index) => value === array2[index])
+
+/*
+ * Here the URL specs:
+
+```
+URL := /simulateur/<category>/<dotted-name>
+
+<category> := 'bilan' // main simulator
+            | 'transport'
+            | 'alimentation'
+            | 'logement'
+            | 'service-sociétaux'
+            | 'divers'
+
+<dotted-name> := // dotted-name with ' . ' replaced by '/' and whitespaces by '-'
+```
+
+Examples:
+
+* The URL corresponding to the question `logement . saisie habitants`
+	of the main simulator is `/simulateur/bilan/logement/saisie-habitants`
+* The URL corresponding to the question  `logement . électricité . consommation`
+  of the sub-simulator `logement` is `/simulateur/logement/électricité/consommation`. Indeed, we only want to access to the sub rules corresponding to the category of the sub-simulator.
+*/
+const availableCategories = [
+	'bilan',
+	'transport',
+	'alimentation',
+	'logement',
+	'services-sociétaux',
+	'divers',
+]
+
+const Simulateur = () => {
+	const dispatch = useDispatch()
+	const urlParams = useParams()
+	const path = urlParams['*']?.split('/')
```

Comment:
```suggestion
	const pathElements = urlParams['*']?.split('/')
```
Pour laisser comprendre que c'est un array avec plusieurs éléments ?